### PR TITLE
Unit tests for multiprocessing

### PR DESCRIPTION
**Pull request by seowwj**
Adds Cpp unit test for multiprocessing

### CODE REVIEW

File: src/lava/magma/runtime/message_infrastructure/test/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+add_executable(
+  test_multiprocessing
+  test_multiprocessing.cc
+)
+
+target_link_libraries(
+  test_multiprocessing
+  GTest::gtest_main
+  pybind11::embed
+  message_infrastructure
```
Comment (hongdami) [inline src/lava/magma/runtime/message_infrastructure/test/CMakeLists.txt:24]: I build error, fixed by adding "lt" to target_link_libraries

File: src/lava/magma/runtime/message_infrastructure/test/test_multiprocessing.cc [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+};
+
+void Builder::Build(int i) {
+  std::cout << "Builder running build " << i << std::endl;
+  std::cout << "Build " << i << "... Sleeping for 10s" << std::endl;
+  sleep(10);
+  std::cout << "Build " << i << "... Builder complete" << std::endl;
+}
+
+AbstractActor::TargetFn TargetFunction(Builder builder, int idx) {
```
Comment (hongdami) [inline src/lava/magma/runtime/message_infrastructure/test/test_multiprocessing.cc]: should be "void(builder, idx, Abstract)"

File: src/lava/magma/runtime/message_infrastructure/test/test_multiprocessing.cc [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+TEST(TestMultiprocessing, MultiprocessingSpawn) {
+  // Spawns an actor
+  // Checks that actor is spawned successfully
+  Builder *builder = new Builder();
+  MultiProcessing mp;
+
+  for (int i = 0; i < 5; i++) {
+    std::cout << "Loop " << i << std::endl;
+    auto bound_fn = std::bind(TargetFunction, std::placeholders::_1, i);
+    mp.BuildActor(bound_fn(*builder));
```
Comment (hongdami) [inline src/lava/magma/runtime/message_infrastructure/test/test_multiprocessing.cc]: placeholders is for actor.

**Review by hongdami — CHANGES_REQUESTED** [review]
I have listed the points which need change. You can try to fix, if feel difficult, you can refer the branch "https://github.com/hongdami/lava/tree/wenjie_test"

File: src/lava/magma/runtime/message_infrastructure/test/CMakeLists.txt (hunk 1 above)
Comment (seowwj) [reply src/lava/magma/runtime/message_infrastructure/test/CMakeLists.txt:24]: Added librt to fix the build issues in the CI as well.
We also need to add exceptions to GoogleTest... it's currently failing linting and security tests

File: src/lava/magma/runtime/message_infrastructure/test/test_multiprocessing.cc [hunk 4]
```diff
@@ -56,7 +56,7 @@ TEST(TestMultiprocessing, MultiprocessingSpawn) {
   std::cout << "Actor List Length --> " << actorList.size() << std::endl;
   for (auto actor : actorList){
     int actorStatus = actor->GetStatus();
-    EXPECT_EQ(actorStatus, 0);
+    EXPECT_EQ(actorStatus, ActorStatus::StatusRunning);
```
Comment (hongdami) [inline src/lava/magma/runtime/message_infrastructure/test/test_multiprocessing.cc]: I think this cannot get the right status.

**Review by hongdami — COMMENTED** [review]
Parent process may cannot get the right actor status for that

**Pull request closed without merging**